packages/go_router/lib/src/route.dart

```diff
@@ -55,169 +45,97 @@ class GoRoute {
-  ///             );
-  ///           },
-  ///           routes: <GoRoute>[
-  ///             GoRoute(
-  ///               path: 'person/:pid',
-  ///               pageBuilder: (BuildContext context, GoRouterState state) {
-  ///                 final Family family = Families.family(state.params['fid']!);
-  ///                 final Person person = family.person(state.params['pid']!);
-  ///                 return MaterialPage<void>(
-  ///                   key: state.pageKey,
-  ///                   child: PersonPage(family: family, person: person),
-  ///                 );
-  ///               },
-  ///             ),
-  ///           ],
-  ///         ),
-  ///       ],
-  ///     ),
-  ///   ],
-  /// );
-  ///
-  /// If there are multiple routes that match the location, the first match is used.
-  /// To make predefined routes to take precedence over dynamic routes eg. '/:id'
-  /// consider adding the dynamic route at the end of the routes
-  /// For example:
-  /// ```
-  /// final GoRouter _router = GoRouter(
-  ///   routes: <GoRoute>[
-  ///     GoRoute(
-  ///       path: '/',
-  ///       redirect: (_) => '/family/${Families.data[0].id}',
-  ///     ),
-  ///     GoRoute(
-  ///       path: '/family',
-  ///       pageBuilder: (BuildContext context, GoRouterState state) => ...,
-  ///     ),
-  ///     GoRoute(
-  ///       path: '/:username',
-  ///       pageBuilder: (BuildContext context, GoRouterState state) => ...,
-  ///     ),
-  ///   ],
-  /// );
-  /// ```
-  /// In the above example, if /family route is matched, it will be used.
-  /// else /:username route will be used.
-  final List<GoRoute> routes;
-
-  /// An optional redirect function for this route.
-  ///
-  /// In the case that you like to make a redirection decision for a specific
-  /// route (or sub-route), you can do so by passing a redirect function to
-  /// the GoRoute constructor.
-  ///
-  /// For example:
-  /// ```
-  /// final GoRouter _router = GoRouter(
-  ///   routes: <GoRoute>[
-  ///     GoRoute(
-  ///       path: '/',
-  ///       redirect: (_) => '/family/${Families.data[0].id}',
-  ///     ),
-  ///     GoRoute(
-  ///       path: '/family/:fid',
-  ///       pageBuilder: (BuildContext context, GoRouterState state) => ...,
-  ///     ),
-  ///   ],
-  /// );
-  /// ```
-  final GoRouterRedirect redirect;
+  /// Extract the path parameters from a match.
+  Map<String, String> extractPathParams(RegExpMatch match) =>
+      extractPathParameters(_pathParams, match);
 
   /// Match this route against a location.
   RegExpMatch? matchPatternAsPrefix(String loc) =>
       _pathRE.matchAsPrefix(loc) as RegExpMatch?;
 
-  /// Extract the path parameters from a match.
-  Map<String, String> extractPathParams(RegExpMatch match) =>
-      extractPathParameters(_pathParams, match);
+  static String? _emptyRedirect(GoRouterState state) => null;
+}
+
+/// Route configuration object equivalent to [StackedRoute].
+class GoRoute extends StackedRoute {
```

Comment:
you can just use typedef if it is just an alias

Reply:
typedefs only work for functions, not classes in Dart, right?

Reply:
I does work on classes since dart 2.13
https://dart.dev/guides/language/language-tour#typedefs

Reply:
Marking as resolved since we are removing StackedRoute.